src/cargo/ops/cargo_add/mod.rs

```diff
@@ -313,8 +313,7 @@ fn resolve_dependency(
             selected
         } else {
             let source = crate::sources::PathSource::new(&path, src.source_id()?, config);
-            let packages = source.read_packages()?;
-            let package = infer_package(packages, &src)?;
+            let package = source.read_packages()?.pop().expect("at least one package");
```

Comment:
Please explain the "expect" the "why", like "`read_packages` errors when no packages"

Reply:
Added!